### PR TITLE
Move coordinator's code into the general module's code directories [MOD-7479]

### PR DESCRIPTION
**Pull request by alonre24**
**Describe the changes in the pull request**

Move the coordinator's source code into the module's source directory, while removing the internal `cmakelists` file in the `coord/src` directory (that had been flattened). The rest of the hierarchy inside `coord` dir stays as is (for now). There are no more changes in this PR.

TODO in upcoming PR(s):
- Move the coordinator's unit tests into the module's `test` directory, and `ramp.yml` into the module's `pack` directory. 
- Build all unit tests as part of the new module's build flow (that is, including coordinator).
- Have a single build path from the module's root dir,
- Simplify debug commands trigger flows (not required of coordinator source is integral in the module)  .

**Mark if applicable**

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

### CODE REVIEW

File: src/coord/rmr/test/CMakeLists.txt [hunk 1]
```diff
@@ -1,8 +1,8 @@
 
-get_filename_component(root ${CMAKE_CURRENT_SOURCE_DIR}/../../../.. ABSOLUTE)
+get_filename_component(root ../../../.. ABSOLUTE)
 message("# coord/rmr/test: root: " ${root})
 
-include_directories(..)
```
Comment (GuyAv46) [inline src/coord/rmr/test/CMakeLists.txt:5]: why not move all the tests into the root `tests` directory?

File: src/coord/pack/ramp.yml
Comment (GuyAv46) [inline src/coord/pack/ramp.yml:1]: Can we move the pack file to the root `pack` directory? I don't think we need this sub-directory

**Review by GuyAv46 — COMMENTED** [review]
Nice
I think we can move all the `coord` sub-directories into the corresponding general sub-directory (tests, pack...)

**Review by GuyAv46 — APPROVED** [review]

File: src/coord/rmr/test/CMakeLists.txt (hunk 1 above)
Comment (alonre24) [reply src/coord/rmr/test/CMakeLists.txt:5]: In next PR

File: src/coord/pack/ramp.yml
Comment (alonre24) [reply src/coord/pack/ramp.yml:1]: In next PR

**Review by DvirDukhan — APPROVED** [review]

**Pull request merged**